bin/reth/src/db/mod.rs

```diff
@@ -92,13 +93,8 @@ impl Command {
         // add network name to data dir
         let data_dir = self.datadir.unwrap_or_chain_default(self.chain.chain);
         let db_path = data_dir.db_path();
-        std::fs::create_dir_all(&db_path)?;
 
-        // TODO: Auto-impl for Database trait
-        let db = reth_db::mdbx::Env::<reth_db::mdbx::WriteMap>::open(
-            db_path.as_ref(),
-            reth_db::mdbx::EnvKind::RW,
-        )?;
+        let db = init_db(&db_path)?;
```

Comment:
Would really like to make this DB::open somehow

Reply:
valid point, I think it's better to address it in a separate refactoring PR